src/plugins/installer.py

```diff
@@ -0,0 +1,92 @@
+import os
+import pathlib
+
+from helperFunctions.install import read_package_list_from_file, run_cmd_with_logging
+
+
+class AbstractPluginInstaller:
+    # Even if some functoins don't need self we want to have them nicely
+    # grouped in this class
+    # pylint:disable=no-self-use
+
+    skip_docker_env = os.getenv('FACT_INSTALLER_SKIP_DOCKER') is not None
+
+    def __init__(self, base_path: pathlib.Path, distribution, skip_docker=skip_docker_env):
+        self.base_path = base_path
+        self.distribution = distribution
+        self.build_path = base_path / 'build'
+        self.skip_docker = skip_docker
+
+    def install(self):
+        cwd = os.getcwd()
+        os.chdir(self.base_path)
+        self.install_system_packages()
+        self.install_pip_packages()
+        self.install_other_packages()
+
+        self.install_files()
+
+        self.build_path.mkdir(exist_ok=True)
+        os.chdir(self.build_path)
+        self.build()
+        run_cmd_with_logging(f'sudo rm -rf {self.build_path}')
```

Comment:
We should add a try-finally block to prevent artifacts from failed installations:
```suggestion
        try:
            self.build_path.mkdir(exist_ok=True)
            os.chdir(self.build_path)
            self.build()
        finally:
            run_cmd_with_logging(f'sudo rm -rf {self.build_path}')
```